src/main/java/ch/njol/skript/expressions/ExprMidpoint.java

```diff
@@ -0,0 +1,146 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Example;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.log.runtime.SyntaxRuntimeErrorProducer;
+
+@Name("Midpoint")
+@Description("Get the midpoint between two vectors or two locations in the same world.")
+@Example("""
+	set {_center} to the midpoint between location(0, 0, 0) and location(10, 10, 10)
+	set {_centerBlock} to the block at {_center}
+	""")
+@Example("set {_midpoint} to the mid-point of vector(20, 10, 5) and vector(3, 6, 9)")
+@Since("INSERT VERSION")
+public class ExprMidpoint extends SimpleExpression<Object> implements SyntaxRuntimeErrorProducer {
+
+	private static final int NONE = 0, LOCATION = 1, VECTOR = 2, BOTH = 3;
```

Comment:
enum please (or just make checkExpressionType return a Class array, that'd probably be easier overall)